SignalMessaging/utils/OWSPreferences.m

```diff
@@ -224,11 +230,41 @@ - (void)setIsSystemCallLogEnabled:(BOOL)flag
 // Therefore in versions of iOS after 11, we have no need of call privacy.
 #pragma mark Legacy CallKit
 
+// Be a little conservative with system call logging with legacy users, even though it's
+// not synced to iCloud, users could be concerned to suddenly see caller names in their
+// recent calls list.
+- (void)applyCallLoggingSettingsForLegacyUsersWithTransaction:(YapDatabaseReadWriteTransaction *)transaction
+{
+    NSNumber *_Nullable callKitPreference =
+        [self tryGetValueForKey:OWSPreferencesKeyCallKitEnabled transaction:transaction];
+    BOOL wasUsingCallKit = callKitPreference ? [callKitPreference boolValue] : YES;
+
+    NSNumber *_Nullable callKitPrivacyPreference =
+        [self tryGetValueForKey:OWSPreferencesKeyCallKitPrivacyEnabled transaction:transaction];
+    BOOL wasUsingCallKitPrivacy = callKitPrivacyPreference ? callKitPrivacyPreference.boolValue : YES;
+
+    BOOL shouldLogCallsInRecents = ^{
+        if (wasUsingCallKit && !wasUsingCallKitPrivacy) {
```

Comment:
Per a conversation with jlund, we should be more conservative about which legacy users we opt in to using the "recents" list. 

Notably, people who left these settings as their default will have to manually opt-in to showing calls in the Recents list.